### PR TITLE
Set Assume Role Session, UserAgent on request to EC2 & fix dirty git tag 

### PR DESCRIPTION
**Pull request by abhipth**
*Issue #, if available:* #37  

*Description of changes:*

Set the User Agent: `amazon-vpc-resource-controller-k8s/git-version <existing user-agent>`
Role Session Name: `amazon-vpc-resource-controller-k8s`

**After Change**
User Agent: `amazon-vpc-resource-controller-k8s/v1.0.11-11-gdc91e0b aws-sdk-go/1.40.43 (go1.16.8; linux; amd64)`
User name `amazon-vpc-resource-controller-k8s`

CloudTrail
```
"userIdentity": {
        "type": "AssumedRole",
        "principalId": "REDACTED:amazon-vpc-resource-controller-k8s",
        "arn": "arn:aws:sts::REDACTED:assumed-role/VPCResourceControllerRole-10/amazon-vpc-resource-controller-k8s",
        "accountId": "REDACTED",
        "sessionContext": {
            "sessionIssuer": {
                "type": "Role",
                "principalId": "REDACTED",
                "arn": "arn:aws:iam::REDACTED:role/VPCResourceControllerRole-10",
                "accountId": "REDACTED",
                "userName": "VPCResourceControllerRole-10"
            },
            "webIdFederationData": {},
            "attributes": {
                "creationDate": "2021-09-16T00:41:56Z",
                "mfaAuthenticated": "false"
            }
        },
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/utils/httpClient.go [hunk 1]
```diff
@@ -20,8 +20,12 @@ import (
 	"golang.org/x/time/rate"
 )
 
+var (
```
Comment (M00nF1sh) [inline pkg/utils/httpClient.go:23]: seems this is too intrusive to the http libarary.
The standard way to add user-agent into aws clients with aws-sdk-go is `https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/aws/user_agent.go#L11`

Comment (abhipth) [reply pkg/utils/httpClient.go:23]: Thanks! I was looking for something like this.

**Review by M00nF1sh — APPROVED** [review]
/lgtm

**Review by orsenthil — APPROVED** [review]
:-)

**Pull request merged**